config/initializers/f_foreman_settings_general.rb

```diff
@@ -39,7 +39,8 @@
       type: :string,
       description: N_('Set an HTTP(s) proxy for all outgoing HTTP(S) connections from Foreman. System-wide proxies must be configured at the operating system level.'),
       default: nil,
-      full_name: N_('HTTP(S) proxy'))
+      full_name: N_('HTTP(S) proxy'),
+      encrypted: true)
```

Comment:
@ekohl do we want to always encrypt the proxy setting?

Reply:
I would certainly not include it in this PR but give it its own PR.

Reply:
@ShimShtein when shouldn't we encrypt it?

Reply:
As long as the proxy setting value does not contain a password, there is no need to encrypt the value.
For example, if I set the proxy to `https://proxy.example.com` there is no need to encrypt it, but if the value is `https://user:pa$$word@proxy.example.com` we have to encrypt it. 
Since we know that the value here is a URL, we can test if the password is there and decide to encrypt based on it.
Since it's not trivial, Ewoud suggests to always encrypt as a first step and then create a separate PR to handle the selective encryption.